src/langs/vala/Compiler.js

```diff
@@ -1,39 +1,33 @@
 import Gio from "gi://Gio";
 import GLib from "gi://GLib";
 import dbus_previewer from "../../Previewer/DBusPreviewer.js";
+import { decode } from "../../util.js";
 
-export default function Compiler(data_dir) {
-  const code_file = Gio.File.new_for_path(
-    GLib.build_filenamev([data_dir, "code.vala"]),
-  );
-  const module_file = Gio.File.new_for_path(
-    GLib.build_filenamev([data_dir, "libworkbenchcode.so"]),
-  );
+export default function Compiler({ session }) {
```

Comment:
@lw64 ok so basically the idea is that each window is now a "session" with its own "session" folder under `~/.var/app/re.sonny.Workbench.Devel/data/re.sonny.Workbench.Devel/sessions/`

Therefor the compiler now scopes everything to the current session folder.